libs/sql-schema-describer/src/postgres.rs

```diff
@@ -389,35 +405,42 @@ impl SqlSchemaDescriber {
             -- we only consider stuff out of one specific schema
             AND tableInfos.relnamespace = schemaInfo.oid
             AND schemaInfo.nspname = $1
-            AND tableInfos.relname = $2
         GROUP BY tableInfos.relname, indexInfos.relname, rawIndex.indisunique, rawIndex.indisprimary
         "#;
         debug!("Getting indices: {}", sql);
         let rows = self
             .conn
-            .query_raw(&sql, &[schema.into(), table_name.into()])
+            .query_raw(&sql, &[schema.into()])
             .await
             .expect("querying for indices");
-        let mut pk: Option<PrimaryKey> = None;
-
-        let mut indices = Vec::new();
 
         for index in rows {
             debug!("Got index: {:?}", index);
             let is_pk = index
                 .get("is_primary_key")
                 .and_then(|x| x.as_bool())
                 .expect("get is_primary_key");
-            // TODO: Implement and use as_slice instead of into_vec, to avoid cloning
+            let table_name = index
+                .get("table_name")
+                .and_then(|x| x.to_string())
+                .expect("get table_name");
+            // TODO: Implement in quaint and use as_slice instead of into_vec, to avoid cloning
             let columns = index
                 .get("column_names")
                 .and_then(|x| x.clone().into_vec::<String>())
                 .expect("column_names");
+
             if is_pk {
-                pk = Some(self.infer_primary_key(schema, table_name, columns, sequences).await);
+                let pk = Some(self.infer_primary_key(schema, &table_name, columns, sequences).await);
```

Comment:
This is the last query that will happen once per table. This may be fixable too, but I haven't looked into it yet.